test/tools/ossfuzz/proto_to_yul.cpp

```diff
@@ -0,0 +1,234 @@
+	{
+		switch (x.op()) {
+			case BinaryOp::ADD:
+				os << "add(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::SUB:
+				os << "sub(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::MUL:
+				os << "mul(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::DIV:
+				os << "div(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::MOD:
+				os << "mod(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::XOR:
+				os << "xor(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::AND:
+				os << "and(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::OR:
+				os << "or(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::EQ:
+				os << "eq(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::LT:
+				os << "lt(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::GT:
+				os << "gt(" << x.left() << "," << x.right() << ")";
+				break;
+		}
+		return os;
+	}
+
+	std::ostream &operator<<(std::ostream &os, const VarDecl &x)
+	{
+		os << "let x_" << x.id() << " := " << x.expr() << "\n";
+		return os;
+	}
+
+	std::ostream &operator<<(std::ostream &os, const TypedVarDecl &x)
+	{
+		os << "let x_" << x.id();
+		switch (x.type())
+		{
+			case TypedVarDecl::BOOL:
+				os << ": bool := " << x.expr() << " : bool\n";
+				break;
+			case TypedVarDecl::S8:
+				os << ": s8 := " << x.expr() << " : s8\n";
+				break;
+			case TypedVarDecl::S32:
+				os << ": s32 := " << x.expr() << " : s32\n";
+				break;
+			case TypedVarDecl::S64:
+				os << ": s64 := " << x.expr() << " : s64\n";
+				break;
+			case TypedVarDecl::S128:
+				os << ": s128 := " << x.expr() << " : s128\n";
+				break;
+			case TypedVarDecl::S256:
+				os << ": s256 := " << x.expr() << " : s256\n";
+				break;
+			case TypedVarDecl::U8:
+				os << ": u8 := " << x.expr() << " : u8\n";
+				break;
+			case TypedVarDecl::U32:
+				os << ": u32 := " << x.expr() << " : u32\n";
+				break;
+			case TypedVarDecl::U64:
+				os << ": u64 := " << x.expr() << " : u64\n";
+				break;
+			case TypedVarDecl::U128:
+				os << ": u128 := " << x.expr() << " : u128\n";
+				break;
+			case TypedVarDecl::U256:
+				os << ": u256 := " << x.expr() << " : u256\n";
+				break;
+		}
+		return os;
+	}
+
+	std::ostream &operator<<(std::ostream &os, const UnaryOp &x)
+	{
+		switch (x.op()) {
+			case UnaryOp::NOT:
+				os << "not(" << x.operand() << ")";
+				break;
+			case UnaryOp::MLOAD:
+				os << "mload(" << x.operand() << ")";
+				break;
+			case UnaryOp::SLOAD:
+				os << "sload(" << x.operand() << ")";
+				break;
+		}
+		return os;
+	}
+
+	std::ostream &operator<<(std::ostream &os, const AssignmentStatement &x)
+	{
+		return os << x.lvalue() << " := " << x.rvalue() << "\n";
+	}
+
+	std::ostream &operator<<(std::ostream &os, const IfStmt &x)
+	{
+		return os << "if " << x.cond() << " {\n"
```

Comment:
If there is a linebreak in a sequence of elements separated by the same separator, there has to be a line break after each separator. This makes it easy to count the number of elements.

Reply:
I'm not sure I follow. Could you please provide an example?